### PR TITLE
feat(typescript): support multiple output targets with declarations

### PR DESCRIPTION
**Pull request by benmccann**
## Rollup Plugin Name: `typescript`

This PR contains:

- [ ] bugfix
- [X] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [X] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [X] no

List any relevant issue numbers:
Closes https://github.com/rollup/plugins/issues/247

### Description

Please see the description in https://github.com/rollup/plugins/issues/247

### CODE REVIEW

File: packages/typescript/src/index.ts [hunk 1]
```diff
@@ -125,11 +125,12 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
         const output = findTypescriptOutput(ts, parsedOptions, fileName, emittedFiles, tsCache);
         output.declarations.forEach((id) => {
           const code = getEmittedFile(id, emittedFiles, tsCache);
-          if (!code) return;
+          const baseDir = outputOptions.dir || (tsconfig && tsconfig.substring(0, tsconfig.lastIndexOf('/')));
```
Comment (NotWoods) [inline packages/typescript/src/index.ts:128]: I think this is a little hard to read, I'd rather just use a few if statements.
```suggestion
          let baseDir = outputOptions.dir;
          if (!baseDir && tsconfig) {
            baseDir = tsconfig.substring(0, tsconfig.lastIndexOf('/'));
          }
```

**Review by NotWoods — APPROVED** [review]
Thank you!

File: packages/typescript/src/index.ts (hunk 1 above)
Comment (benmccann) [reply packages/typescript/src/index.ts]: done. thanks for the quick review!

**Comment by shellscape** [comment]
thanks!

**Pull request merged**